spectate/spectate.py

```diff
@@ -201,12 +202,49 @@ def expose(base, *methods, **kwargs):
     return type(name, (base, WatchableType), classdict)
 
 
+# TODO: Proofread doc string.
+def expose_as(name, base, *methods):
+    """Expose a type as a WatchableType with methods that are exposed callbacks.
+
+    Parameters
+    ----------
+    name : str
+    base : obj:
+        A type such as list or dict.
+    methods : str
+        A str representation of the methods to expose.
+
+    Returns
+    -------
+    exposed : obj:
+        A WatchableType in the form of `spectate.spectate.name`
+    """
+    exposed = expose(base, *methods, name=name)
+    return exposed
+
+
+# TODO: Proofread doc string.
 def watchable(value):
+    """Return a boolean if value class is WatchableType."""
     check = issubclass if inspect.isclass(value) else isinstance
     return check(value, WatchableType)
 
 
+# TODO: Proofread doc string.
 def watch(value, *args, **kwargs):
+    """Return a WatchableType with it's spectator from a WatchableType.
+
+    In order to register callbacks to an eventful object, you need to create
+    a Spectator that will watch it for you. A Spectator is a relatively simple
+    object that has methods for adding, deleting, and triggering callbacks. To
+    create a spectator we call spectator = watch(x), where x is a
+    WatchableType.
+
+    Parameters
+    ----------
+    value : obj:
+        WatchableType
+    """
     if inspect.isclass(value):
         value = value(*args, **kwargs)
         return value, watch(value)
```

Comment:
@rmorshea I have another question. Is there any reason to have `watch` return a tuple here? Could both the spectator and the watchable object be returned in some kind of container class? Actually could the the spectator as a attribute of the watchable type?

Reply:
> Could both the spectator and the watchable object be returned in some kind of container class?

They could be returned in some sort of container class, but I feel like, the `tuple` serves that function equally well without requiring some new construct.

> Could the spectator could be set as an attribute?

Yes, and it already is. They key thing to realize though, it that it must be set on one specific attribute because this is where `MethodSpectator` descriptors look for it when they try to notify. In my eyes this is not a public API, but a private part of how callbacks get triggered under the hood. So obfuscating these details behind a function reduces confusion, and makes it possible to do a bit of helpful type checking. It also allows some freedom in the future in case we come up with a better way for a `MethodSpectator` to communicate with a `Spectator`.